File: jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        stones_counter = Counter(stones)
        count = 0
        
        for jewel in jewels:
            if jewel in stones_counter:
                count += stones_counter[jewel]
        
        return count
            

File: jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> jewelSet = new HashSet<>();
        int count = 0;
        
        for(char jewel : jewels.toCharArray()){
            jewelSet.add(jewel);
        }
        
        for(char stone : stones.toCharArray()){
            if(jewelSet.contains(stone)){
                count += 1;
            }
        }
        
        return count;
    }
}